app/services/data_sources/json_setting_data_source.rb

```diff
@@ -0,0 +1,99 @@
+require "fileutils"
+require_relative "json_data_source"
+require_relative "setting_data_source"
+require_relative "../../shared/logging_module"
+require_relative "../../shared/json_convertible"
+require_relative "../../shared/models/setting"
+
+module FastlaneCI
+  # Mixin the JSONConvertible class for Setting
+  class Setting
+    include FastlaneCI::JSONConvertible
+  end
+
+  # Data source for settings variables backed by JSON
+  class JSONSettingDataSource < SettingDataSource
+    include FastlaneCI::JSONDataSource
+    include FastlaneCI::Logging
+
+    class << self
+      attr_accessor :file_semaphore
+    end
+
+    # can't have us reading and writing to a file at the same time
+    # TODO: currently we just have this global lock
+    #   Instead we'll have multiple instances of `JSONSettingDataSource` objects
+    #   for each project and a global one
+    JSONSettingDataSource.file_semaphore = Mutex.new
+
+    def after_creation(**params)
+      FileUtils.mkdir_p(json_folder_path)
+      logger.debug("Using folder path for settings data: #{json_folder_path}")
+      # load up the json file here
+      # parse all data into objects so we can fail fast on error
+      reload_settings
+    end
+
+    def setting_file_path(path: "settings.json")
+      return File.join(json_folder_path, path)
+    end
+
+    # TODO: make synchronize work again
+    def settings
+      # JSONSettingDataSource.file_semaphore.synchronize do
```

Comment:
does this not work? It looks like it's used in line 49.

Reply:
Yes, worked when I tested it 👍 Added an open todo for when we work on the file sync and git commits